strictdoc/helpers/ordered_set.py

```diff
@@ -48,5 +48,13 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f"<OrderedSet {self}>"
 
-    def sort(self) -> None:
-        self._d = dict(sorted(self._d.items()))
+    def sort(self, key=None) -> None:
```

Comment:
I didn't implement this in my branch, so my comparator function when sorting OrderedSet was giving me tuples, not objects. You will have to fix this in one place here:

```
        # Sort by node UID alphabetically.
        for path_requirements_ in self.map_paths_to_reqs.values():

            def node_comparator(node) -> int:
                return assert_cast(node[0].reserved_uid, str)

            path_requirements_.sort(key=node_comparator)
```